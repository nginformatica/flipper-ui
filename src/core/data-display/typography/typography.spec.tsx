import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { Typography } from '.'

describe('Typography', () => {
    it('should render with no variant', () => {
        render(<Typography>Some text</Typography>)

        const typography = screen.getByText('Some text')

        expect(typography).toBeDefined()
        expect(typography.classList).toContain('MuiTypography-body2')
    })

    it('should render with variant', () => {
        render(<Typography variant='h1'>Title</Typography>)

        const typography = screen.getByText('Title')

        expect(typography).toBeDefined()
        expect(typography.classList).toContain('MuiTypography-h1')
    })

    it('should render with custom style', () => {
        render(
            <Typography margin={10} padding={5} style={{ color: 'red' }}>
                Some text
            </Typography>
        )

        const typography = screen.getByText('Some text')

        expect(typography.style.color).toBe('red')
        expect(typography.style.margin).toBe('10px')
        expect(typography.style.padding).toBe('5px')
    })
})
