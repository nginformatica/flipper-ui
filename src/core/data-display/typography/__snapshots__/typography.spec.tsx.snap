// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should match snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body2"
    style="margin: 10px; padding: 5px; color: red;"
  >
    Some text
  </p>
</div>
`;
